core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -406,8 +408,69 @@ impl TransactionViewReceiveAndBuffer {
 
         let mut num_received = 0usize;
         let mut num_buffered = 0usize;
+        let mut num_dropped_on_status_age_checks = 0usize;
         let mut num_dropped_on_capacity = 0usize;
         let mut num_dropped_on_receive = 0usize;
+
+        // Create temporary batches of transactions to be age-checked.
+        let mut transaction_priority_ids = ArrayVec::<_, EXTRA_CAPACITY>::new();
+        let lock_results: [_; EXTRA_CAPACITY] = core::array::from_fn(|_| Ok(()));
+        let mut error_counters = TransactionErrorMetrics::default();
+
+        let mut check_and_push_to_queue =
+            |container: &mut TransactionViewStateContainer,
+             transaction_priority_ids: &mut ArrayVec<TransactionPriorityId, 64>| {
+                // Temporary scope so that transaction references are immediately
+                // dropped and transactions not passing
+                let check_results = {
+                    let mut transactions = ArrayVec::<_, EXTRA_CAPACITY>::new();
+                    transactions.extend(transaction_priority_ids.iter().map(|priority_id| {
+                        &container
+                            .get_transaction_ttl(priority_id.id)
+                            .expect("transaction must exist")
+                            .transaction
+                    }));
+                    working_bank.check_transactions::<RuntimeTransaction<_>>(
+                        &transactions,
+                        &lock_results[..transactions.len()],
+                        MAX_PROCESSING_AGE,
+                        &mut error_counters,
+                    )
+                };
+
+                // Remove errored transactions
+                for (result, priority_id) in check_results
+                    .iter_mut()
+                    .zip(transaction_priority_ids.iter())
+                {
+                    if result.is_err() {
+                        num_dropped_on_status_age_checks += 1;
+                        container.remove_by_id(priority_id.id);
+                    }
+                    let transaction = &container
+                        .get_transaction_ttl(priority_id.id)
+                        .expect("transaction must exist")
+                        .transaction;
+                    *result = Consumer::check_fee_payer_unlocked(
+                        &working_bank,
+                        transaction,
+                        &mut error_counters,
+                    );
+                    if result.is_err() {
+                        num_dropped_on_status_age_checks += 1;
+                        container.remove_by_id(priority_id.id);
+                    }
+                }
+                // Push non-errored transaction into queue.
```

Comment:
@jstarry, functional change here to apply similar change as #4865.
In commit: c5fc7f63cecd119726bee56815ebdc5686ed355f

Reply:
Great, looks like the error type is mismatched FYI (CI failure). And we should update the name of `num_dropped_on_status_age_checks` or add a separate counter for txs dropped due to invalid fee payer

Reply:
yeah it should be renamed, keeping it in the same one for now since that's what the other PR does. Will separate the metric in a separate PR, changing for both ingest paths.

Reply:
> Great, looks like the error type is mismatched FYI (CI failure). And we should update the name of `num_dropped_on_status_age_checks` or add a separate counter for txs dropped due to invalid fee payer

ugh yeah - pushed to CI for testing since the rocksdb update was causing a bunch of issues in my dev environment yesterday w/ not finding libclang. Can actually build now. Should have just been patient and waited, sorry for the waste of time reviewing that!